src/components/Layout/RightPaneLayout/useRightPane.tsx

```diff
@@ -0,0 +1,17 @@
+import { OpenRightPaneOpts, useRightPaneContext } from "./RightPaneContext";
+
+export interface UseRightPaneHook {
+  /** Opens a right pane */
+  openRightPane: (opts: OpenRightPaneOpts) => void;
+  /** Closes the right pane */
+  closeRightPane: () => void;
+}
+
+export function useRightPane(): UseRightPaneHook {
+  const { openInPane, closePane } = useRightPaneContext();
+
+  return {
+    openRightPane: (opts: OpenRightPaneOpts) => openInPane(opts),
+    closeRightPane: () => closePane(),
+  };
```

Comment:
**suggestion:** I know pushed you towards not using `onCallback` for `openRightPane` and `closeRightPane` in my previous suggestions, but that was because I was thinking we could directly reference the `openInPane` and `closePane` methods, vs wrapping them in another function to call them. As it is right now the `openRightPane` and `closeRightPane` methods will be considered "new" whenever the calling component re-renders, which can have some side-effects of unnecessary recalculations.
Consider:
```
function MyComponent() {
  const { openRightPane } = useRightPane();
  
  const myCallback = useCallback((id) => {
    openRightPane(id);
  }, [openRightPane]);
  
  const [hovered, setHover] = useHover();

  return (
    <div 
      onClick={myCallback(123)}
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
      css={Css.if(hover).underline.$}
   >
      Click Me!
   </div>
  );
}
```
Whenever we hover over the div, it triggers the component to rerender, reevaluating the `useRightPane` hook, getting a new function/object in memory, which then triggers the `useCallback` to reevaluate, generating a new `myCallback` in memory as well. 

So TLDR: Let's either directly reference the stable `openInPane` and `closePane` methods
```
return { openRightPane: openInPane, closeRightPane: closePane }
```
Or memoize the local methods so they don't trigger unnecessary rerenders downstream.

Sorry if this was over verbose.

Reply:
yeah makes sense to just pass them down